5th task of testing

IvayloTsvetkov opened the pull request:
SMS app

mgenov commented on testing_5/src/com/company/SMSSender.java:
```diff
@@ -0,0 +1,7 @@
+package com.company;
+
+public class SMSSender {
+  boolean sendSMS(SMS sms, SMSVerifier smsVerifier) {
```
> Поради факта, че външната система не е реализирана, приложението трябва да може да работи и в тестов режим, който да му позволява да работи без външната система за разпращане на съобщенията 

e.g SMSGateway is missing. You have to mock and the external system which is used for delivering of the messages. 

Currently you are verifying that message is just valid and not that it's sent to the gateway for processing.

mgenov commented on testing_5/test/com/company/SMSSenderTest.java:
```diff
@@ -0,0 +1,36 @@
+public class SMSSenderTest {
+  @Rule public JUnitRuleMockery context = new JUnitRuleMockery();
+  @Mock SMSVerifier smsVerifier;
+  SMS sms = new SMS("0888123456", "Random subject", "Hello");
```
Move sms in the test. 

You can improve name of the reference: anyShortMessage = new SMS("") to better describe that this message data is not related with the test and any instance could be passed.

mgenov commented on testing_5/src/com/company/SMS.java:
```diff
@@ -0,0 +1,26 @@
+    this.message = message;
+  }
+
+  boolean isMessageLengthValid() {
```
You are not having tests for the validation of the SMS, e.g SMSValidator's implementation is missing. 

Currently these methods for validation here are not used.

IvayloTsvetkov commented:
Added implementation and tests for SMSVerifier and SMSGateway.

mgenov commented:
You cannot have tests of SMSGateway as you are not having real SMS gateway.

mgenov commented on testing_5/test/com/company/SMSGatewayTest.java:
```diff
@@ -0,0 +1,43 @@
+  SMSGateway smsGateway = new SMSGateway(smsSender, verifier);
+
+  @Test
+  public void sendSMSSuccessful() {
```
You have missing case when message is not valid.

mgenov commented on testing_5/src/com/company/Main.java:
```diff
@@ -0,0 +1,8 @@
+package com.company;
+
+public class Main {
+  public static void main(String[] args) {
```
App should use SMSGateway and all of it's collaborators.

mgenov commented on testing_5/test/com/company/SMSVerifierTest.java:
```diff
@@ -0,0 +1,40 @@
+  }
+
+  @Test
+  public void SubjectIsMissing() {
```
methods should start with lower case character.

IvayloTsvetkov commented:
Changed first letter of SubjectIsMissing() to lowercase, added sendSMSWithInvalidMessage() test and made SMSGateway use all it's collaborators in the Main class by overriding Sender's method to return true explicitly.
